input/otlp/metrics.go

```diff
@@ -75,8 +83,13 @@ func (c *Consumer) ConsumeMetrics(ctx context.Context, metrics pmetric.Metrics)
 	if err := c.config.Processor.ProcessBatch(ctx, batch); err != nil {
 		return ConsumeMetricsResult{}, err
 	}
+	var errMsg string
+	if remainingDataPoints > 0 {
+		errMsg = "unsupported data points"
+	}
 	return ConsumeMetricsResult{
 		RejectedDataPoints: remainingDataPoints,
+		ErrorMessage:       errMsg,
```

Comment:
Overloading this PR to add one missing element in #156 , which is to return an error message along with the rejected count. According to the spec, it MUST return a rejected count and SHOULD return a message. For the sake of simplicity and that we do not have any guarantees about the stability of the `Consume*WithResult` interface, we'll just have a string in `Consume*Result` struct instead of private fields + public methods.

In reality, currently, this ErrorMessage field is only non-empty when metrics rejected count > 0. The "unsupported data points" is not extremely helpful, but still better than nothing.